### PR TITLE
Collection counters

### PR DESCRIPTION
**Pull request by Mikearaya**
for #312

### CODE REVIEW

File: packages/core-assortments/db/helpers.js [hunk 1]
```diff
@@ -110,6 +110,32 @@ Collections.Assortments.findAssortments = ({
   return Collections.Assortments.find(selector, options).fetch();
 };
 
+Collections.Assortments.count = async ({
```
Comment (pozylon) [inline packages/core-assortments/db/helpers.js:113]: Super nice you have added these count functions to the collections but please extract the selector building code to minimize duplication and reduce likelihood of wrong counts when just one selector got changed in one function.

Kind of like:
```
const buildFindSelector = (props) => ({ isActive: ..., ... }); // contains selector building code of findAssortments
Assortments.findAssortments = (props) { sort=..., Assortments.find(buildFindSelector(props), sort).fetch() }
Assortments.count = (props) { Assortments.find(buildFindSelector(props)).count() }
```

**Review by pozylon — CHANGES_REQUESTED** [review]
I found some already existing count fields and I think we should unify the naming and deprecate the old ones for these:

```
ype ProductCollection {
      totalCount: Int!
      filteredCount: Int!
      items: [Product!]!
    }

type LoadedFilterOption {
      filteredProducts: Int!
      definition: FilterOption!
      isSelected: Boolean
    }

type LoadedFilter {
      examinedProducts: Int!
      filteredProducts: Int!
      definition: Filter!
      isSelected: Boolean
      options: [LoadedFilterOption!]
    }

"""
    Search result
    """
    type SearchResult {
      totalProducts: Int!
      filteredProducts: Int!
      filters: [LoadedFilter!]!
      products(limit: Int = 10, offset: Int = 0): [Product!]!
    }

    """
    Search result
    """
    type ProductSearchResult {
      totalProducts: Int!
      filteredProducts: Int!
      filters: [LoadedFilter!]!
      products(limit: Int = 10, offset: Int = 0): [Product!]!
    }

    type AssortmentSearchResult {
      totalAssortments: Int!
      assortments(limit: Int = 10, offset: Int = 0): [Assortment!]!
    }
```

For ex. the last one could be:

```
type AssortmentSearchResult {
      totalCount: Int! @deprecated(reason: "Renamed, use the assortmentCount field")
      assortmentCount: Int!
      assortments(limit: Int = 10, offset: Int = 0): [Assortment!]!
    }
```

File: packages/core-assortments/db/helpers.js (hunk 1 above)
Comment (Mikearaya) [reply packages/core-assortments/db/helpers.js:113]: @pozylon while i agree making the field names uniform is good i think some of the fields that you outlined above are descriptive and shouldn't be deprecated like  `filteredProducts` while `examinedProducts` can be replaced with `productsCount`

Comment (pozylon) [reply packages/core-assortments/db/helpers.js]: Agreed, makes sense

File: package.json [hunk 2]
```diff
@@ -20,7 +20,7 @@
   },
   "scripts": {
     "lint": "eslint packages tests --cache --fix --ext=js --ext=ts",
-    "pretest": "npm run lint --silent && npm run typecheck",
```
Comment (pozylon) [inline package.json:23]: why that? we should make sure that the code is linted

**Review by pozylon — CHANGES_REQUESTED** [review]

File: package.json (hunk 2 above)
Comment (Mikearaya) [reply package.json:23]: @pozylon typescript fails on locally that why i remove it and forgot, putting it back.

**Comment by Mikearaya** [comment]
@pozylon this is ready

File: package.json (hunk 2 above)
Comment (pozylon) [reply package.json:23]: @Mikearaya ping

**Review by pozylon — APPROVED** [review]

**Comment by Mikearaya** [comment]
@pozylon :raised_hands:

**Pull request merged**